test/integration/util/shell.go

```diff
@@ -0,0 +1,298 @@
+// +build integration
+
+/*
+Copyright (C) 2017 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"bufio"
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"os/exec"
+	"strings"
+
+	"github.com/DATA-DOG/godog/gherkin"
+	"github.com/minishift/minishift/pkg/util/os"
+)
+
+const (
+	exitCodeIdentifier = "exitCodeOfLastCommandInShell="
+
+	bashExitCodeCheck       = "echo %v$?"
+	fishExitCodeCheck       = "echo %v$status"
+	tcshExitCodeCheck       = "echo %v$?"
+	zshExitCodeCheck        = "echo %v$?"
+	cmdExitCodeCheck        = "echo %v%%errorlevel%%"
+	powershellExitCodeCheck = "echo %v$lastexitcode"
+)
+
+var (
+	shell shellInstance
```

Comment:
OK, shell is maintained from start till exit.

Reply:
@gbraad Unfortunately it is, however the shell instance is being created and removed during that time.

Reply:
why unfortunately... this is actually what you want to happen...